Library/Homebrew/brew.sh

```diff
@@ -373,7 +373,7 @@ setup_curl() {
      "${HOMEBREW_BREWED_CURL_PATH}" --version &>/dev/null
   then
     HOMEBREW_CURL="${HOMEBREW_BREWED_CURL_PATH}"
-  elif [[ -n "${HOMEBREW_DEVELOPER}" && -x "${HOMEBREW_CURL_PATH}" ]]
+  elif [[ -n "${HOMEBREW_CURL_PATH}" ]]
```

Comment:
```suggestion
  elif [[ -x "${HOMEBREW_CURL_PATH}" ]]
```

Do we still want to check whether the path is executable?

Reply:
@Rylan12 I don't think so; we don't have good enough messaging or error-handling in that case and silently ignoring something set by the user seems likely to cause more confusion than it helps.